src/libfuncs.rs

```diff
@@ -270,14 +274,14 @@ where
     'this: 'ctx,
 {
     #[doc(hidden)]
-    pub(crate) fn results(self) -> impl Iterator<Item = Vec<Value<'ctx, 'this>>> {
+    pub(crate) fn results(self) -> impl Iterator<Item = NativeResult<Vec<Value<'ctx, 'this>>>> {
```

Comment:
```rs
pub(crate) fn results(self) -> impl Iterator<Item = NativeResult<Vec<Value<'ctx, 'this>>>> {
```
If you're going to do this, then returning `Result<impl Iterator<...>>` would be better. Like this:
```rs
pub(crate) fn results(self) -> NativeResult<impl Iterator<Item = Vec<Value<'ctx, 'this>>>> {
    // ...
}
```

Reply:
Yes, I first tried to tackle it that way. The problem is that the error might happen inside the `map()` and since the iterator expects not a `Result<Vec<Value<'ctx, 'this>>` but a ` Vec<Value<'ctx, 'this>`, it breaks.

Reply:
Aah, that's true. As long as the iterator is lazy we won't be able to move up the result. Could we make it eager? Usually libfuncs don't have that many result branches, right?

Reply:
Yes, please make it eager.